tinyagentos/routes/agent_debugger.py

```
@@ -0,0 +1,226 @@
+async def debugger_trace(agent_id: str, request: Request) -> JSONResponse:
+    """Record a trace event from the agent runtime.
+
+    Agents (or adapters) POST here to push tool call / result / prompt events
+    into the trace buffer. The SSE stream picks them up and broadcasts to
+    connected debugger UIs.
+
+    Request body: {"type": "tool_call"|"tool_result"|"prompt", ...}
+    """
+    try:
+        body = await request.json()
+    except Exception:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)